File: client/src/application/lessons/dto/lesson_opening_content.dto.js
class LessonOpeningContentDTO {
  constructor({
    lesson_id,
    lesson_number,
    lesson_name,
    status,
    opened_at,
    closed_at,
  }) {
    this.lessonId = lesson_id ?? "";
    this.lessonNumber = lesson_number ?? 0;
    this.lessonName = lesson_name ?? "";
    this.status = status ?? "locked";
    this.openedAt = opened_at ?? null;
    this.closedAt = closed_at ?? null;
  }

  static fromResponse(data) {
    return new LessonOpeningContentDTO(data || {});
  }
}

module.exports = LessonOpeningContentDTO;

File: server/apps/lessons/application/dto/lesson_opening_content_dto.py
from dataclasses import dataclass
from datetime import datetime


@dataclass
class LessonOpeningContentDTO:

    lesson_opening_id: str

    university_id: str
    university_name: str

    class_section_lesson_plan_id: str

    lesson_id: str
    lesson_number: int
    lesson_name: str

    status: str

    opened_at: datetime | None
    closed_at: datetime | None

    created_at: datetime
    updated_at: datetime